mlx/backend/cuda/compiled.cpp

```diff
@@ -205,28 +249,19 @@ void Compiled::eval_gpu(
     builder.os += "\n} // namespace mlx::core::cu\n";
     // Build kernel names.
     std::vector<std::string> kernel_names;
-    for (auto work_per_thread : std::array<int, 2>{1, 4}) {
+    for (auto wpt : std::array<int, 2>{1, work_per_thread}) {
       kernel_names.push_back(fmt::format(
-          "mlx::core::cu::{}_contiguous<uint32_t, {}>",
-          lib_name(),
-          work_per_thread));
+          "mlx::core::cu::{}_contiguous<uint32_t, {}>", lib_name(), wpt));
       kernel_names.push_back(fmt::format(
-          "mlx::core::cu::{}_contiguous<int64_t, {}>",
-          lib_name(),
-          work_per_thread));
+          "mlx::core::cu::{}_contiguous<int64_t, {}>", lib_name(), wpt));
```

Comment:
I don't think we need the `work_per_thread = 1` for the contiguous kernels? Like you could move the start of this loop down a couple lines and only build the contiguous kernels for `work_per_thread`?